assets/js/admin/promote-job-modals.js

```diff
@@ -0,0 +1,49 @@
+/**
```

Comment:
This logic was extracted from the `admin.js`, so we can use it properly in the editor.